File: backend/spring-boot/src/main/java/org/bootstrapbugz/api/auth/payload/request/AuthTokensRequest.java
package org.bootstrapbugz.api.auth.payload.request;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import org.bootstrapbugz.api.shared.constants.Regex;
import org.bootstrapbugz.api.shared.validator.UsernameOrEmail;

public record AuthTokensRequest(
    @NotBlank(message = "{user.usernameOrEmailRequired}")
        @UsernameOrEmail(message = "{user.usernameOrEmailInvalid}")
        String usernameOrEmail,
    @NotBlank(message = "{user.passwordRequired}")
        @Pattern(regexp = Regex.PASSWORD, message = "{user.passwordInvalid}")
        String password) {}

File: backend/spring-boot-kotlin/src/main/kotlin/org/bootstrapbugz/api/auth/payload/request/AuthTokensRequest.kt
package org.bootstrapbugz.api.auth.payload.request

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Pattern
import org.bootstrapbugz.api.shared.constants.Regex
import org.bootstrapbugz.api.shared.validator.UsernameOrEmail

data class AuthTokensRequest(
    @UsernameOrEmail(message = "{auth.invalid}")
    @NotBlank(message = "{usernameOrEmail.required}")
    val usernameOrEmail: String?,
    @NotBlank(message = "{password.required}")
    @field:Pattern(regexp = Regex.PASSWORD, message = "{auth.invalid}")
    val password: String?
)
